### PR TITLE
Task5

### PR DESCRIPTION
**Pull request by bluluke**
Return all article objects instead of only article objects with comments

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -82,3 +83,28 @@ describe('GET /api/articles/:article_id', () => {
...
+        expect(article).toHaveProperty("author", expect.any(String));
+        expect(article).toHaveProperty("title", expect.any(String));
+        expect(article).toHaveProperty("article_id", expect.any(Number));
+        expect(article).toHaveProperty("topic", expect.any(String));
+        expect(article).toHaveProperty("created_at", expect.any(String));
+        expect(article).toHaveProperty("votes", expect.any(Number));
+        expect(article).toHaveProperty("article_img_url", expect.any(String));
+        expect(article).toHaveProperty("comment_count", expect.any(String));
+  
+      })
```
Comment (Liam310) [inline __tests__/app.test.js:107]: Great test! 👌

File: db/controllers/allArticlesController.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+
+const { selectAllOfArticles } = require('../models/allArticlesModel');
+
+exports.getAllArticles = (req, res, next) => {
+  selectAllOfArticles().then((data) => {
+    res.status(200).send(data);
+  })
```
Comment (Liam310) [inline db/controllers/allArticlesController.js:7]: The common convention when serving data is not just to respond with an array of the data, but to wrap it in an object and put it on a descriptive property (like in this case, `articles`)

File: db/models/allArticlesModel.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+const db = require('../connection');
+
+exports.selectAllOfArticles = (req) => {
+
+    return db
+        .query("WITH article_cols AS (SELECT articles.author, articles.title, articles.article_id, articles.topic, articles.created_at, articles.votes, articles.article_img_url FROM articles), comment_count AS (SELECT article_id, COUNT(*) AS comment_count FROM comments GROUP BY article_id), article_comment_count AS (SELECT article_cols.author, article_cols.title, COALESCE(article_cols.article_id, (SELECT article_id FROM articles WHERE article_cols.title = articles.title)) AS article_id, article_cols.topic, article_cols.created_at, article_cols.votes, article_cols.article_img_url, COALESCE(comment_count.comment_count, '0') AS comment_count FROM article_cols LEFT JOIN comment_count ON article_cols.article_id = comment_count.article_id) SELECT * FROM article_comment_count ORDER BY created_at DESC;")
```
Comment (Liam310) [inline db/models/allArticlesModel.js:6]: Naturally we just discussed this plenty! But just so you have a written reference for when you want to think about a refactor, all the pieces of the puzzle are here, and the ones that matter are:
- `SELECT`ing from the articles table
- `LEFT JOIN`ing the comments table
- `COUNT` and `GROUP BY` to get the comment_count

**Review by Liam310 — APPROVED** [review]
Thank you for taking the time to explain your work to me, Luke.

Go ahead and merge your work - the refactor can come later on but right now, it all works 🙌

File: db/models/allArticlesModel.js (hunk 3 above)
Comment (bluluke) [reply db/models/allArticlesModel.js:6]: Thanks for feedback!

**Pull request merged**